packages/lesmis/src/components/DashboardReportModal.js

```diff
@@ -0,0 +1,152 @@
+/*
+ * Tupaia
+ * Copyright (c) 2017 - 2020 Beyond Essential Systems Pty Ltd
+ *
+ */
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+import styled from 'styled-components';
+import { useTheme } from '@material-ui/core/styles';
+import ToggleButtonGroup from '@material-ui/lab/ToggleButtonGroup';
+import { GridOn, KeyboardArrowRight, BarChart } from '@material-ui/icons';
+import {
+  useMediaQuery,
+  Slide,
+  Typography,
+  Dialog as MuiDialog,
+  Container as MuiContainer,
+  Button as MuiButton,
+} from '@material-ui/core';
+import * as COLORS from '../constants';
+import { FlexSpaceBetween, FlexEnd, FlexStart } from './Layout';
+import { DialogHeader } from './FullScreenDialog';
+import { ToggleButton } from './ToggleButton';
+import { YearSelector } from './YearSelector';
+import { ChartTable, TABS } from './ChartTable';
+
+const Transition = React.forwardRef(function Transition(props, ref) {
+  return <Slide direction="left" ref={ref} {...props} />;
+});
+
+const Wrapper = styled.div`
+  height: 100%;
+  overflow: auto;
+  background: ${COLORS.GREY_F9};
+`;
+
+const Container = styled(MuiContainer)`
+  padding: 1.25rem 6.25rem 6.25rem;
+  padding-bottom: 10vh;
+  height: 100%;
+  display: flex;
+  flex-direction: column;
+`;
+
+const Header = styled(FlexSpaceBetween)`
+  min-height: 5.625rem;
+  border-bottom: 1px solid ${props => props.theme.palette.grey['400']};
+  margin-bottom: 1.8rem;
+
+  .MuiTextField-root {
+    margin-right: 0;
+  }
+`;
+
+const Heading = styled(Typography)`
+  font-size: 1.25rem;
+  line-height: 1.4rem;
+  font-weight: 500;
+`;
+
+export const DashboardReportModal = ({
+  name,
+  dashboardGroupName,
+  buttonText,
+  entityCode,
+  dashboardGroupId,
+  reportId,
+  periodGranularity,
+  year,
+}) => {
+  const [open, setOpen] = useState(false);
+  const [selectedYear, setSelectedYear] = useState(year);
+  const theme = useTheme();
+  const fullScreen = useMediaQuery(theme.breakpoints.down('sm'));
```

Comment:
Wow this is cool

Reply:
I just discovered it yesterday - it seems like it could be very useful for sure.